### PR TITLE
Include the ccx_keys package when launching tasks

### PR DESCRIPTION
**Pull request by bdero**
Closes https://openedx.atlassian.net/browse/AN-5666.

This adds ccx_keys to the list of modules that are included in tarballs whenever the HadoopTaskRunner-based mapreduce engine is used. This is necessary for any CCX course event logs to be processed by any mapreduce tasks.
### Testing instructions:
1. Setup an analytics devstack: https://github.com/edx/edx-documentation/blob/813c5cd23034ac0fc171f7f9b6ab70051b0b5eac/en_us/install_operations/source/installation/analytics/install_analytics.rst#installing-analytics-devstack
2. Checkout this branch: `git checkout bdero/ccx-keys`
3. Create a CCX and add enroll a user in it from within edX: http://edx.readthedocs.io/projects/open-edx-ca/en/latest/set_up_course/custom_courses.html
4. Run the ImportEnrollmentsIntoMysql:
   
   ``` bash
   remote-task --host localhost --user hadoop --remote-name analyticstack --skip-setup --wait ImportEnrollmentsIntoMysql --interval 2016 --local-scheduler
   ```
5. Check the course_enrollment_daily report table to observe that enrollment information for the CCX is in the table (the password for user `read_only` is `password`): 
   
   ``` bash
   echo "select id, course_id, created from reports.course_enrollment_daily group by course_id;" | mysql -u read_only -p
   ```

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @bdero! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fedx-analytics-pipeline&number=290

**Comment by pomegranited** [comment]
:+1:  It does what it says on the box! :smile:

Please make the 2 minor changes listed to the task description before creating the OSPR.
- [x] I tested this using the above instructions, with small modifications:
  - Step 4: use `ImportEnrollmentsIntoMysql` task:
    
    ```
    remote-task --host localhost --user hadoop --remote-name analyticstack --skip-setup --wait ImportEnrollmentsIntoMysql --interval 2016 --local-scheduler
    ```
  - Step 5: Need to provide the database name:
    
    ```
    echo "select id, course_id, created from course_enrollment_daily group by course_id;" | mysql -u read_only -p reports
    ```
- [x] I read through the code
- [x] Includes documentation - in comments.

**Comment by bdero** [comment]
Thanks @pomegranited , I updated the PR Description. This is ready for upstream review.

**Comment by mulby** [comment]
Can we make this configurable?

Other than that it LGTM.

**Comment by bdero** [comment]
@mulby Sure, would pulling in a config entry from the `event-logs` key be sufficient?

I.e.

``` ini
[event-logs]
pattern = .*tracking.log.*
expand_interval = 2 days
source = hdfs://localhost:9000/data/
process_ccx_logs = true
```

**Comment by mulby** [comment]
I think it warrants a separate section since it's not really event-log specific:

```
[ccx]
enabled = true
```

**Comment by bdero** [comment]
@mulby Alright, I used that as the config value and rebased. This is ready for further review!

File: edx/analytics/tasks/launchers/local.py [hunk 1]
```diff
@@ -56,9 +56,14 @@ def main():
     # - cjson is used for all parsing event logs.
     # - filechunkio is used for multipart uploads of large files to s3.
     # - opaque_keys is used to interpret serialized course_ids
+    #   - opaque_keys extensions:  ccx_keys
     #   - dependencies of opaque_keys:  bson, stevedore
     luigi.hadoop.attach(boto, cjson, filechunkio, opaque_keys, bson, stevedore, ciso8601, requests)
 
+    if configuration.getboolean('ccx', 'enabled', default=True):
```
Comment (mulby) [inline edx/analytics/tasks/launchers/local.py]: Can we make this default to `False`? I suspect most installations won't use this feature.

**Review by mulby — CHANGES_REQUESTED** [review]

File: edx/analytics/tasks/launchers/local.py (hunk 1 above)
Comment (bdero) [reply edx/analytics/tasks/launchers/local.py]: Done.

**Review by mulby — APPROVED** [review]

**Comment by bdero** [comment]
Thanks for approving @mulby - I don't have permission to merge this PR, however. Otherwise I would. :)

**Comment by pomegranited** [comment]
Hey @bdero  - do you want to squash your commits?  I can merge when you're ready.

**Comment by bdero** [comment]
@pomegranited Alrighty, all squashed. Thanks.

**Pull request merged**